### PR TITLE
Cleanup: remove ZZZ folder and lodash dep.

### PR DESCRIPTION
**Pull request by nicolad**
## Problem this Pull Request solves
- While investigating this issue: https://github.com/eventespresso/event-espresso-core/issues/2931
I came upon an issue related to lodash versions conflicts and decided to replace lodash with ramda's equivalents from 2 files where it was used:
- `assets/src/application/ui/layout/espressoTable/utils.ts`
- `assets/src/application/ui/layout/espressoTable/TableHeader.tsx`
- Also found some other few references in `ZZZ` folder and decided to remove that since we have that copy anyway in `EDTR/refactor-master`

## How has this been tested
- Locally, by running both `yarn dev` and `yarn prod` and then checking table view, since that utility was used for table.

### CODE REVIEW

File: assets/src/application/ui/layout/espressoTable/TableHeader.tsx [hunk 1]
```diff
@@ -1,6 +1,5 @@
 import React from 'react';
 import classNames from 'classnames';
-import { isFunction } from 'lodash';
```
Comment (nicolad) [inline assets/src/application/ui/layout/espressoTable/TableHeader.tsx:3]: lodash was used here

File: assets/src/application/ui/layout/espressoTable/utils.ts [hunk 2]
```diff
@@ -1,18 +1,19 @@
 import classNames from 'classnames';
 import { Children } from 'react';
-import { castArray, filter, first, isArray, isEmpty, isFunction, last, reject } from 'lodash';
```
Comment (nicolad) [inline assets/src/application/ui/layout/espressoTable/utils.ts:3]: and also here

File: assets/src/application/ui/layout/espressoTable/utils.ts [hunk 3]
```diff
@@ -60,15 +61,6 @@ export const isTableRow = (element) => {
...
 
-export const isArrayOfTableCells = (elements) => {
-	elements = castArray(elements);
-	let allAreCells = true;
-	elements.forEach((cell) => {
-		allAreCells = isTableHeaderCell(cell) || isTableDataCell(cell) ? allAreCells : false;
-	});
-	return allAreCells;
-};
-
```
Comment (nicolad) [inline assets/src/application/ui/layout/espressoTable/utils.ts:71]: I've searched through `assets/src` and this function isn't used

File: assets/src/application/ui/layout/espressoTable/utils.ts [hunk 4]
```diff
@@ -98,19 +90,3 @@ export const addZebraStripesOnMobile = (exclude) => (cells) => {
...
- * @param {Array} columns
- * @param {string} exclude table row cell "key"
- * @return {Array} columns
- */
-export const filterColumnsByKey = (columns, exclude) => {
-	if (!Array.isArray(columns)) {
-		return columns;
-	}
-	return exclude ? reject(columns, ['key', exclude]) : columns;
-};
```
Comment (nicolad) [inline assets/src/application/ui/layout/espressoTable/utils.ts:116]: same here, searched through `assets/src` and this function isn't used

File: assets/vendor/lodash.js [hunk 5]
```diff
@@ -1,17107 +0,0 @@
-/**
```
Comment (nicolad) [inline assets/vendor/lodash.js:1]: I don't think we need this, do we?

File: assets/vendor/lodash.min.js [hunk 6]
```diff
@@ -1,137 +0,0 @@
-/**
```
Comment (nicolad) [inline assets/vendor/lodash.min.js:1]: same here, do we need this?

File: tests/mocks/addons/eea-new-addon/package.json [hunk 7]
```diff
@@ -68,7 +68,6 @@
 		"@wordpress/i18n": "^1.1.0",
 		"@wordpress/jest-preset-default": "^1.0.5",
 		"classnames": "^2.2.5",
-		"lodash": "4.17.4",
```
Comment (nicolad) [inline tests/mocks/addons/eea-new-addon/package.json:71]: removed here

File: tests/mocks/addons/new-messages-template-pack-variation/package.json [hunk 8]
```diff
@@ -68,7 +68,6 @@
 		"@wordpress/i18n": "^1.1.0",
 		"@wordpress/jest-preset-default": "^1.0.5",
 		"classnames": "^2.2.5",
-		"lodash": "4.17.4",
```
Comment (nicolad) [inline tests/mocks/addons/new-messages-template-pack-variation/package.json:71]: here removed

File: tests/mocks/addons/new-messages-template-pack/package.json [hunk 9]
```diff
@@ -67,7 +67,6 @@
 		"@wordpress/i18n": "^1.1.0",
 		"@wordpress/jest-preset-default": "^1.0.5",
 		"classnames": "^2.2.5",
-		"lodash": "4.17.4",
```
Comment (nicolad) [inline tests/mocks/addons/new-messages-template-pack/package.json:70]: removed

File: tests/mocks/addons/new-payment-method/package.json [hunk 10]
```diff
@@ -67,7 +67,6 @@
 		"@wordpress/i18n": "^1.1.0",
 		"@wordpress/jest-preset-default": "^1.0.5",
 		"classnames": "^2.2.5",
-		"lodash": "4.17.4",
```
Comment (nicolad) [inline tests/mocks/addons/new-payment-method/package.json:70]: removed here

**Comment by nicolad** [comment]
@tn3rb I noticed some `package.json` files in few locations under `tests/mocks/addons/.../package.json`, are those still being used anywhere?

**Comment by tn3rb** [comment]
>  I noticed some package.json files in few locations under tests/mocks/addons/.../package.json, are those still being used anywhere?

i'm not exactly sure why those were ever added, but I think they were intended as templates for new add-ons... why they are in the tests directory I don't know, It wasn't always easy to get straight answers out of some previous devs.

anyways, if they are not being used anywhere then I'm ok with them being removed.

**Pull request merged**